src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -1167,8 +1167,12 @@ namespace OpenLoco::Vehicles
 
             status = Status::waitingAtSignal;
 
+            auto newConfig = Config::getNew();
             auto* vehType1 = train.veh1;
-            vehType1->timeAtSignal++;
+            if (newConfig.trains_reverse_at_signals)
+            {
+                vehType1->timeAtSignal++;
+            }
```

Comment:
Conceptually, I would have liked the waiting time check to be removed rather than the increment. However, thinking about this, that would mean `timeAtSignal` could eventually overflow. I therefore prefer your approach :-)

Reply:
The thought did cross my mind while testing. Just wondering now if we could check the config separately before incrementing and returning `true` where required. Don't know if that's a good idea though...

eg.

```cpp
...
    status = Status::waitingAtSignal;

    auto& newConfig = Config::getNew();
    if (newConfig.trainsReverseAtSignals)
    {
        return true;
    }

    auto* vehType1 = train.veh1;
    vehType1->timeAtSignal++;
...
```

Reply:
@duncanspumpkin Any thoughts about whether returning early would be better?